### PR TITLE
perf(auth): validate user through cache to avoid duplicate hash calculation

### PR DESCRIPTION
**Pull request by imbajin**
Consider use `MurmurHash3` (later)

### CODE REVIEW

File: hugegraph-core/src/main/java/com/baidu/hugegraph/auth/StandardAuthManager.java [hunk 1]
```diff
@@ -341,8 +345,17 @@ public HugeUser matchUser(String name, String password) {
         E.checkArgumentNotNull(name, "User name can't be null");
         E.checkArgumentNotNull(password, "User password can't be null");
         HugeUser user = this.findUser(name);
-        if (user != null &&
-            StringEncoding.checkPassword(password, user.password())) {
+        if (user == null) {
+            return null;
+        }
+
+        Id id = IdGenerator.of(name);
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/auth/StandardAuthManager.java:352]: use user.id()

File: hugegraph-core/src/main/java/com/baidu/hugegraph/auth/StandardAuthManager.java [hunk 2]
```diff
@@ -45,6 +45,7 @@
     private final HugeGraphParams graph;
     private final EventListener eventListener;
     private final Cache<Id, HugeUser> usersCache;
+    private final Cache<Id, String> validateCache;
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/auth/StandardAuthManager.java:48]: rename to pwdCache

File: hugegraph-core/src/main/java/com/baidu/hugegraph/auth/StandardAuthManager.java [hunk 3]
```diff
@@ -59,6 +60,8 @@ public StandardAuthManager(HugeGraphParams graph) {
         this.graph = graph;
         this.eventListener = this.listenChanges();
         this.usersCache = this.cache("users");
+        this.validateCache = CacheManager.instance().cache("validate_pwd");
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/auth/StandardAuthManager.java:63]: this.cache("users_pwd")

**Review by javeme — CHANGES_REQUESTED** [review]

File: hugegraph-core/src/main/java/com/baidu/hugegraph/auth/StandardAuthManager.java (hunk 3 above)
Comment (imbajin) [reply hugegraph-core/src/main/java/com/baidu/hugegraph/auth/StandardAuthManager.java:63]: `cache(string)`  returns `Cache<Id, HugeUser>`, but we need `Cache<Id, String>` now

**Review by javeme — APPROVED** [review]

**Review by Linary — APPROVED** [review]

**Review by zhoney — APPROVED** [review]

**Pull request merged**